### ISSUE TITLE
[DAP] add inline values of variables during debugging

### ISSUE THREAD

**Issue by filipdutescu** [helix-editor/helix#5927]
# Description

Similar to IntelliJ, VS Code, PyCharm etc. show values of variables used on each line during debugging. This should occur during a pause of the session (ie breakpoint was hit, waiting for next step into/out/over etc.). The values should probably only de shown for the current scope (the scope in which debugging is paused).

Design-wise, I propose the following alternatives, of which I prefer the first the best.

1: 
![image](https://user-images.githubusercontent.com/46979062/218281674-f6ab070e-8e95-474f-b2a1-8e15180de4ba.png)
2:
![image](https://user-images.githubusercontent.com/46979062/218281686-fccc39f3-d051-43e6-bdf9-1dbc6ed196b8.png)
3:
![image](https://user-images.githubusercontent.com/46979062/218281697-ff518abe-3d53-4d0d-9058-666d22befcc8.png)
4:
![image](https://user-images.githubusercontent.com/46979062/218281713-e48542e5-eb0a-4886-8440-c7c1798facb9.png)

# Proposal

Follow VSCode's approach to it (described in [their documentation][1]), which means using both language servers and debugging adapters, since the latter will probably not support it directly ([source][2]). The general mechanism appears to be:
1. Request the list of variables local to the current scope.
2. Request the values of said variables from the debug adapter
3. Show the received values

From discussions with @pascalkuthe on the Matrix channel, it seems like this approach is pretty new and lacks mass adoption, at the time of writing. However, I still think there is value in implementing it prior to said adoption, to make it so Helix in ready to use it, as soon as it reaches the required parties.

# External references

1. [VSCode's documentation][1]
2. [DAP request for an inline values request][2]
3. [DAP Variables request][3]
4. [LSP Inline values request][4]

[1]: https://github.com/microsoft/vscode-docs/blob/vnext/release-notes/v1_54.md#inline-value-provider-api
[2]: https://github.com/microsoft/debug-adapter-protocol/issues/304
[3]: https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Variables
[4]: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_inlineValue

**Comment by pascalkuthe** [comment]
> From discussions with @pascalkuthe on the Matrix channel, it seems like this approach is pretty new and lacks mass adoption, at the time of writing. However, I still think there is value in implementing it prior to said adoption, to make it so Helix in ready to use it, as soon as it reaches the required parties.

If we can land this feature in helix then I would look into implement the necessary capabilities to rust-alayzer so we can have a great debugging experience for rust atleast.

**Comment by pascalkuthe** [comment]
I agree that the first alternative loks the best. It doesn't get in the way :+1:

**Comment by filipdutescu** [comment]
> If we can land this feature in helix then I would look into implement the necessary capabilities to rust-alayzer so we can have a great debugging experience for rust atleast. I think atleast the eclipse LSP implements this too.
> ...

That's really great! I'll work on the debugging redesign first, as per our talk over on Matrix and then will try to also take a jab at this issue!
